TIM1_IRQn wrong definition for STM32F410x

Hi!
I'm playing around with my Nucleo F410RE board to see if I can get it working with this core.
I've created a generic pinout for it (that I haven't pushed yet), but I'm not able to compile code for it due to a bug (?) in the core system files.

This is the error I get whenever I compile something:
```
In file included from C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c:14:
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c: In function 'getTimerUpIrq':
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/timer.h:54:19: error: 'TIM1_UP_TIM10_IRQn' undeclared (first use in this function); did you mean 'TIM1_BRK_TIM9_IRQn'?
   54 | #define TIM1_IRQn TIM1_UP_TIM10_IRQn
      |                   ^~~~~~~~~~~~~~~~~~
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c:374:16: note: in expansion of macro 'TIM1_IRQn'
  374 |         IRQn = TIM1_IRQn;
      |                ^~~~~~~~~
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\cores\arduino/stm32/timer.h:54:19: note: each undeclared identifier is reported only once for each function it appears in
   54 | #define TIM1_IRQn TIM1_UP_TIM10_IRQn
      |                   ^~~~~~~~~~~~~~~~~~
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c:374:16: note: in expansion of macro 'TIM1_IRQn'
  374 |         IRQn = TIM1_IRQn;
      |                ^~~~~~~~~
```

However: if I change this line in stm32f410rx.h to `TIM1_UP_TIM10_IRQn` it does compile without any errors:
https://github.com/stm32duino/Arduino_Core_STM32/blob/a37f9590e6752076068f893280ee88fe3d456f36/system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f410rx.h#L115

Any reason why `TIM1_UP_TIM10_IRQn` isn't defined for any of the F410 targets?

Well, sadly it didn't solve the problem 😢 

```c
#elif defined(STM32F3xx) || defined(STM32L4xx) || defined(STM32WBxx)
#define TIM1_IRQn TIM1_UP_TIM16_IRQn
#define TIM1_IRQHandler TIM1_UP_TIM16_IRQHandler
#elif defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
#if defined(TIM10_BASE) && !defined(TIM10_IRQn)
#define TIM1_IRQn TIM1_UP_TIM10_IRQn
#define TIM1_IRQHandler TIM1_UP_TIM10_IRQHandler
#endif
#elif defined(STM32H7xx) || defined(STM32MP1xx)
#define TIM1_IRQn TIM1_UP_IRQn
#define TIM1_IRQHandler TIM1_UP_IRQHandler
#endif

```

```
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c: In function 'getTimerUpIrq':
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c:374:16: error: 'TIM1_IRQn' undeclared (first use in this function); did you mean 'TIM5_IRQn'?
  374 |         IRQn = TIM1_IRQn;
      |                ^~~~~~~~~
      |                TIM5_IRQn
C:\Users\user\AppData\Local\Arduino15\packages\STM32\hardware\stm32\1.8.0\libraries\SrcWrapper\src\stm32\timer.c:374:16: note: each undeclared identifier is reported only once for each function it appears in
```

--- a/cores/arduino/stm32/timer.h
+++ b/cores/arduino/stm32/timer.h
@@ -51,8 +51,13 @@ extern "C" {
 #define TIM1_IRQn TIM1_UP_TIM16_IRQn
 #define TIM1_IRQHandler TIM1_UP_TIM16_IRQHandler
 #elif defined(STM32F2xx) || defined(STM32F4xx) || defined(STM32F7xx)
+#if !defined (TIM10_BASE)
+#define TIM1_IRQn TIM1_UP_IRQn
+#define TIM1_IRQHandler TIM1_UP_IRQHandler
+#else
 #define TIM1_IRQn TIM1_UP_TIM10_IRQn
 #define TIM1_IRQHandler TIM1_UP_TIM10_IRQHandler
+#endif
 #elif defined(STM32H7xx) || defined(STM32MP1xx)
 #define TIM1_IRQn TIM1_UP_IRQn
 #define TIM1_IRQHandler TIM1_UP_IRQHandler

